grails-app/migrations/views/cycle-count-metadata.sql

```diff
@@ -0,0 +1,25 @@
+CREATE OR REPLACE VIEW cycle_count_metadata AS
+SELECT
+    CRC32(CONCAT(inventory_id, product_id)) as id,
+    cycle_count_metadata_tmp.inventory_id,
+    cycle_count_metadata_tmp.product_id,
+    cycle_count_metadata_tmp.date_counted,
+    cycle_count_metadata_tmp.days_since_last_count,
+    cycle_count_metadata_tmp.abc_class,
+    cycle_count_frequency.frequency,
+    date_add(cycle_count_metadata_tmp.date_counted, INTERVAL cycle_count_frequency.frequency DAY) as date_expected,
+    datediff(date_add(cycle_count_metadata_tmp.date_counted, INTERVAL cycle_count_frequency.frequency DAY), now()) as days_until_next_count
+FROM (
+         SELECT
+             product_count_history.inventory_id,
+             product_count_history.product_id,
+             product_count_history.date_counted,
+             datediff(now(), date_counted) as days_since_last_count,
+             -- FIXME Used to clean data, this should not be used in the actual solution
+             CASE WHEN abc_class = 'A' THEN 'A' WHEN abc_class = 'B' THEN 'B' WHEN abc_class = 'C' THEN 'C' ELSE 'DEFAULT' END as abc_class
+         FROM product_count_history
+                  LEFT OUTER JOIN product_classification
+                                  ON product_count_history.inventory_id = product_classification.inventory_id
+                                      AND product_count_history.product_id = product_classification.product_id
+     ) as cycle_count_metadata_tmp
+         LEFT OUTER JOIN cycle_count_frequency ON cycle_count_frequency.abc_class = cycle_count_metadata_tmp.abc_class;
```

Comment:
This would benefit from a concrete table for cycle count frequency but still need to test if it's required.

Reply:
I tested whether it helps and it does not so I'm going to leave cycle count frequency as a view for now.